server/routes/api/comments/comments.ts

```diff
@@ -64,15 +64,26 @@ router.post(
     const { sort, direction, documentId } = ctx.input.body;
     const { user } = ctx.state.auth;
 
-    const document = await Document.findByPk(documentId, { userId: user.id });
-    authorize(user, "read", document);
-
-    const comments = await Comment.findAll({
-      where: { documentId },
-      order: [[sort, direction]],
-      offset: ctx.state.pagination.offset,
-      limit: ctx.state.pagination.limit,
-    });
+    let comments;
+    if (documentId) {
+      const document = await Document.findByPk(documentId, { userId: user.id });
+      authorize(user, "read", document);
+      comments = await Comment.findAll({
+        where: { documentId },
+        order: [[sort, direction]],
+        offset: ctx.state.pagination.offset,
+        limit: ctx.state.pagination.limit,
+      });
+    } else {
+      comments = await Comment.scope(["defaultScope", "withDocument"]).findAll({
+        where: {
+          "$document.teamId$": user.teamId,
```

Comment:
I've not seen this syntax before, probably better to use `include` here instead of `withDocument` scope and then you can add the where clause directly on the include.

One other problem here is that although it's limiting to the team, it's not limiting to the collections that the user has access to, so currently comments from private collections will be returned. You can use `user.collectionIds()` to get a list of `collectionIds` to filter to.

Reply:
Sure @tommoor. Thank you. I've done those changes now. Could you take another look when you have time?

Reply:
Looks bood, while we're here you could allow passing a specific `collectionId` to get comments for a specific collection.

Are you adding tests?

Reply:
Great suggestion @tommoor . I've added the logic to get comments for a specific collection using `collectionId`.

I couldn't find any tests for the `comments.list` event. I think I missed them. So I haven't added them. Do you know where I can find the tests? ( I tried global search for `comments.list` but couldn't find any occurrences in a test file )

Reply:
The comments routes are currently missing a test file, but pretty much every other api route has one so there are lots of equivalent tests for `list` endpoints to look at